tests/integration_tests/test_deploy_model.py

```diff
@@ -0,0 +1,88 @@
+import unittest
+import subprocess
+import os
+import requests
+import time
+import platform
+import signal
+import tempfile
+import shutil
+
+
+class TestDeployModel(unittest.TestCase):
+    def __init__(self, *args, **kwargs):
+        super(TestDeployModel, self).__init__(*args, **kwargs)
+        self.cwd = os.getcwd()
+        self.tabpy_root = os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..')
+        if platform.system() == 'Windows':
+            self.py = 'python'
+        else:
+            self.py = 'python3'
+
+    def setUp(self):
+        prefix = '_TestDeployModel_'
+        self.state_dir = tempfile.mkdtemp(prefix=prefix, dir=self.cwd)
+        self.state_file = open(os.path.join(self.state_dir, 'state.ini'), 'w+')
+        self.state_file.write('[Service Info]\n'
+                             'Name = TabPy Serve\n'
+                             'Description = \n'
+                             'Creation Time = 0\n'
+                             'Access-Control-Allow-Origin = \n'
+                             'Access-Control-Allow-Headers = \n'
+                             'Access-Control-Allow-Methods = \n'
+                             '\n'
+                             '[Query Objects Service Versions]\n'
+                             '\n'
+                             '[Query Objects Docstrings]\n'
+                             '\n'
+                             '[Meta]\n'
+                             'Revision Number = 1\n')
+        self.state_file.close()
+
+        self.config_file = tempfile.NamedTemporaryFile(
+            mode='w+t', prefix=prefix, suffix='.conf', dir= self.cwd, delete=False)
+        self.config_file.write('[TabPy]\n'
+            'TABPY_PORT= 9004\n'
+            'TABPY_STATE_PATH = {}'.format(self.state_dir))
+        self.config_file.close()
+
+        
+        os.chdir(self.tabpy_root)
+        # start TabPy server in the background
+        if platform.system() == 'Windows':
+            self.process = subprocess.Popen(['startup.cmd', os.path.basename(self.config_file.name), '&'])
+        else:
+            self.process = subprocess.Popen(['./startup.sh', '--config=' + os.path.basename(self.config_file.name), '&'],
+                                             preexec_fn=os.setsid)
+        time.sleep(1)
+
+    def tearDown(self):
```

Comment:
Do you want to use the classmethod here? Seems pretty expensive to kill the TabPy server on every test instance

Reply:
Note that this is equivalent for now since there is only one test, but if you add more tests killing the serve each time seems like a lot

Reply:
Thanks for all of your feedback, it's been very helpful especially in regards to python3. 

We actually want to start and stop a server for every test since we don't want any failures related to models to affect the test runs for other features. We also have somewhat specific tabpy environments for the tests we're in the process of adding so it wouldn't be possible to use the same server for each of those tests.

Reply:
Gotcha. So what I would suggest is that every environment you want to configure would be its own class with class methods to start up and tear down the server appropriately. That would keep things organized and get you the same effect you want. By starting and stopping every test we wouldn't really be mirroring production usage, since it's unrealistic for end users to do the same thing in a production environment after every endpoint gets queried.

Again realize that this is easier said than done, so not something that needs to be done in this PR but should be given consideration